domain/storage/service/storagepool.go

```diff
@@ -360,6 +360,20 @@ func (s *StoragePoolService) GetStoragePoolByName(ctx context.Context, name stri
 	return pool, nil
 }
 
+// GetStoragePoolByUUID returns the storage pool with the UUID.
+// The following errors can be expected:
+// - [storageerrors.PoolNotFoundError] if a pool with the specified name does not exist.
+func (s *StoragePoolService) GetStoragePoolByUUID(ctx context.Context, poolUUID domainstorage.StoragePoolUUID) (domainstorage.StoragePool, error) {
```

Comment:
Delete this. We still have a lot of work todo in this area over the coming days and this will force us down a path we may not have time to deal with. All of this can be done in the application domain.